src/lib/schematics/update/tests/class-names/input.ts

```diff
@@ -0,0 +1,14 @@
+import {ConnectedOverlayDirective, OverlayOrigin} from '@angular/cdk/overlay';
```

Comment:
Should we be including the version in here as well? E.g., this test case is based on v5, but that's not captured anywhere. I'm think we could do something like:
```
schematics/update/test-cases/v5/ts-class-name_input.ts
schematics/update/test-cases/v5/ts-class-name_expected_output.ts
```

Reply:
I actually didn't want differentiate between V5 and V6 because otherwise it wouldn't be really verifying how the tool will work in CLI projects that are on V5 and just run `ng update`.

Also it kind of depends on how we handle the V5 and V6 data (e.g. whether we merge the data; or have separate rules; or separate schematics).

I'd personally vote for a follow-up. PR. The PR is already pretty big and is primarily about the infrastructure and ability to run such test cases.

Reply:
Ah, interesting. I had always assumed we would _only_ run the v5 -> v6 rules when the user is _on_ v5, and then the same for v6 -> v7 and so on. Now I realize, though, it's set up to run all of them all the time. 

As far as correctness goes, I don't think it matters; the symbols from v5 should never come back, so the old rules would be a no-op. However, I would be slightly concerned that there would be a performance cost to running all of the rules for old versions (in making `ng update` take longer). I'd want to try running them over an app or two to gauge the cost.

Reply:
Also cc @hansl for input

Reply:
It's worth mentioning as well that we also removed the various NPM installs for the update. So the performance should be better than before anyway.

Reply:
Thinking about it more, sequencing could also become a problem. E.g., if we refactor `PortalHost` -> `PortalOutlet` in v6, and then changed `PortalOutlet` to `ContentOutlet` in v8, we'd have to make sure the fixes ran in the right order. 

I think that the [migration.json](https://github.com/angular/material2/blob/master/src/lib/schematics/migration.json#L5) drives what's run based on the version. Have multiple entry-points based on this wouldn't need to duplicate any code, they would just all call the same function with some `version` parameter

Reply:
That's understandable. I'd just need to find a way to tell the rules what data they should take. That might be a different problem though.

If we go that way (regardless of how we'd be doing it), then having separate folders for the test-cases (like you mentioned) makes sense.
´
Though I feel that having a folder for a test-case would keep the test-cases more structured. e.g. `update/test-cases/v5/class-names/input.ts|expected_output.ts`

Reply:
I don't like have one directory per test case since it's always exactly two files; in my mind it makes more sense to have all the test cases for one version in the same directory. I also wanted to avoid having other files called `input.ts` since it collides with the file for `matInput`

Reply:
@mmalerba @josephperrott @vivian-hu do you have any opinions on file organization here?

Reply:
Makes sense. I will make the change tomorrow.

Reply:
> it wouldn't be really verifying how the tool will work in CLI projects that are on V5 and just run ng update.

The CLI will run all migrations between [from, to] (inclusive), in order. So you will run the v5 -> v6 migration, then whatever is after v6 (I would presume v6 -> v7 migration).

I need to get you off tslint asap. Or you need to get off of it yourself :) . It just complicates things.

Reply:
@hansl I see 😄 I didn't really integrate the TSLint rules into the schematics. At some point the reasoning for the TSLint rules in the initial standalone tool was that the rules could be run inside of Google. But not sure how relevant this is today.

What would you recommend for doing the update tool? Just using the TS API without TSLint?

Reply:
For now, that's unfortunately the best.